### PR TITLE
Add automatic instrumentation for WebRequest.GetResponse() calls that reference System 2.0.0.0

### PR DESCRIPTION
**Pull request by zacharycmontoya**
Changes proposed in this pull request:
Similar to the recent System.Data v2.0.0.0 PR (#1000), we do not currently instrument calls to `WebRequest.GetResponse()` when libraries reference System 2.0.0.0, because we set the minimum version of calls to be 4.0.0.0. This lowers the minimum version allowed and adds an integration test to demonstrate the added instrumentation.

@DataDog/apm-dotnet

### CODE REVIEW

File: test/test-applications/integrations/Samples.WebRequest.NetFramework20/Program.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    // write response content
+                    byte[] responseBytes = Utf8.GetBytes(ResponseContent);
+                    context.Response.ContentEncoding = Utf8;
+                    context.Response.ContentLength64 = responseBytes.Length;
+                    context.Response.OutputStream.Write(responseBytes, 0, responseBytes.Length);
+
+                    // we must close the response
+                    context.Response.Close();
+                }
+                catch (HttpListenerException)
```
Comment (tonyredondo) [inline test/test-applications/integrations/Samples.WebRequest.NetFramework20/Program.cs:130]: I guess we are missing a couple of catch here: https://github.com/DataDog/dd-trace-dotnet/blob/5c37e00cc0059f76c1f43c1d4e55e7dafcf9f4a2/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs#L210-L222

**Review by tonyredondo — APPROVED** [review]
LGTM, I leave one nit.

File: test/test-applications/integrations/Samples.WebRequest.NetFramework20/Program.cs (hunk 1 above)
Comment (zacharycmontoya) [reply test/test-applications/integrations/Samples.WebRequest.NetFramework20/Program.cs:130]: Thanks for pointing out, added in c63d1c59e14a449b68bcfe38805cde8612929e5c

File: integrations.json [hunk 2]
```diff
@@ -2298,7 +2298,7 @@
           "signature_types": [
             "System.Net.WebResponse"
           ],
-          "minimum_major": 4,
+          "minimum_major": 2,
```
Comment (lucaspimentel) [inline integrations.json:2301]: `+551 -17` for this single character change 😅

**Review by lucaspimentel — APPROVED** [review]

**Pull request merged**